File: 49-group-anagrams/49-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> closedColission = new HashMap<>();
        for(String str : strs) {
            char[] strArr = str.toCharArray();
            Arrays.sort(strArr);
            String sorted = String.valueOf(strArr);
            if (!closedColission.containsKey(sorted)) {
                closedColission.put(sorted, new ArrayList<>());
            }
            closedColission.get(sorted).add(str);
        }
        return new ArrayList<>(closedColission.values());
    }
}

File: 49-group-anagrams/49-group-anagrams.py
from math import ceil

class Solution:
    # return hash value
#     def hashFunc(self, phrase, n):
#         hashVal = 10000
#         for l in phrase:
#             hashVal -= ord(l)

#         return hashVal
    
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashMap = {}
        n = len(strs)
        for word in strs:
            # hashVal = self.hashFunc(word, n)
            hashVal = "".join(sorted(word))
            
            if hashVal not in hashMap:
                hashMap[hashVal] = []
            hashMap[hashVal].append(word)

        return hashMap.values()
        